napari/_vispy/canvas.py

```diff
@@ -634,6 +640,64 @@ def _add_overlay_to_visual(self, overlay: Overlay) -> None:
             vispy_overlay.node.parent = self.view.scene
         self._overlay_to_visual[overlay] = vispy_overlay
 
+    def _calculate_view_direction(self, event_pos: List[float]) -> List[float]:
+        """calculate view direction by ray shot from the camera"""
+        # this method is only implemented for 3 dimension
+        if self.viewer.dims.ndisplay == 2:
+            return self.viewer.camera.calculate_nd_view_direction(
+                self.viewer.dims.ndim, self.viewer.dims.displayed
+            )
+        x, y = event_pos
+        w, h = self.size
+        nd = self.viewer.dims.ndisplay
+
+        transform = self.view.scene.transform
+        # map click pos to scene coordinates
+        click_scene = transform.imap([x, y, 0, 1])
+        # canvas center at infinite far z- (eye position in canvas coordinates)
+        eye_canvas = [w / 2, h / 2, -1e10, 1]
+        # map eye pos to scene coordinates
+        eye_scene = transform.imap(eye_canvas)
+        # homogeneous coordinate to cartesian
+        click_scene = click_scene[0:nd] / click_scene[nd]
+        # homogeneous coordinate to cartesian
+        eye_scene = eye_scene[0:nd] / eye_scene[nd]
+
+        # calculate direction of the ray
+        d = eye_scene - click_scene
```

Comment:
This is the problem @jni:

```suggestion
        d = click_scene - eye_scene
```